pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -87,12 +87,12 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	//
 	// TODO: Any changes to a part of the object that represents desired state (labels,
 	// annotations etc) should also increment the generation.
-	if !reflect.DeepEqual(oldDaemonSet.Spec.Template, newDaemonSet.Spec.Template) {
+	if !apiequality.Semantic.DeepEqual(oldDaemonSet.Spec.Template, newDaemonSet.Spec.Template) {
```

Comment:
I thought when we were talking about this, you had said that there was a difference where nil vs [] meant different things. Will this fail to trigger on certain updates that it actually should trigger on?